### DISCUSSION TITLE
How to stop mirthconnect mcserver using external python script after conversion through channels .

### DISCUSSION THREAD

**Discussion by sal4455** [nextgenhealthcare/connect#5150] [General]
Hi,
I have created a script to start the mirth-server-launcher.jar using python script, but I could'nt find a way how to stop the mirth-server-launcher.jar just after conversion of all messages from mirth channels. Please help me out with this .
Is there any mirth functionality available that I can leverage for stopping mirth server automatically after conversion through channel?
Thankyou in Advance .

**Comment by pacmano1** [comment]
Is there a reason you are not using the standard start (systemd) scripts?  

 I assume you are on linux, but you didn't post much about your environment.

"mirth server automatically after conversion through channel?" what does that mean?

**Reply by sal4455** [reply]
Hey I'm sorry if I interpret it wrongly , but By standard start scripts you mean by starting mirth using - **"mcserver start"** or is there any other scripts to start it ?

I want to create a separate script which will start the mirthconnect server then transform all the messages that are in the directory mentioned in the source connector- **"File Writer"** and then stop the server just after transforming all the messages through its channels . Please suggest me some ideas or help me out with this .

Yes I am using linux environment and mirthconnect version 3.11.0 with postgress database.

**Reply by pacmano1** [reply]
Yes.  Mirth server has standard start scripts that you can call like any other systemd processes.  However that is dependent on how you install mirth.  For example the standard installer does install those scripts although it runs mirth as root which is bad practice.  Most folks just create their own systemd startups scripts, you can see forum posts on that.

Why are you stopping the engine after it processes?  That isn't a typical way to run mirth.  Why not just stop the channels and leave mirth running?   

Maybe rewind a bit and give more detail about your use case.

**Reply by sal4455** [reply]
Thankyou @pacmano1  but I want to start mirth server using python script and it is working fine for me .The only blocker that I am facing is to stop either the channel or the server through python script .

Why are you stopping the engine after it processes? -> Because If I don't stop ,channels will receive all the messages/hl7 files whenever any message will be in the particular folder. and I want to make such script which will work - when it start the server ,only when messages should be transformed rather than transforming everytime.

 As you are saying to stop the channel instead of server so Is it possible to stop the channel externally? and if yes so how it is possible ?
Instead of stopping manually, is there any way to stop either the server or channel automatically after transformation of messages?

**Reply by pacmano1** [reply]
Mirth has a REST API to start and stop channels among many other things that it does.   Also, I am not sure you understand Mirth very well, it is specifically designed for things such that you don't need to drive it with other software like Python.

You should really thoroughly describe your use case here.      For example, mirth could be polling a directory every second and skipping files based on some criteria you set somewhere else.  Or it could poll a database and ask "should I collect files now".   Or it can skip files that are not at least a day old.   There are many options.

**Comment by pacmano1** [comment]
Also, this should be in discussions, not posted as an issue, moving it there.

**Reply by pacmano1** [reply]
Please don't cross post to issues and discussions.
